libs/utils/src/properties.c

```diff
@@ -467,11 +469,42 @@ int celix_properties_size(const celix_properties_t *properties) {
 }
 
 celix_properties_iterator_t celix_propertiesIterator_construct(const celix_properties_t *properties) {
-    return hashMapIterator_construct((hash_map_t*)properties);
+    celix_properties_iterator_t iter;
```

Comment:
```suggestion
    celix_properties_iterator_t iter;
    CELIX_BUILD_ASSERT(sizeof(celix_properties_iterator_t) == sizeof(hash_map_iterator_t));
    CELIX_BUILD_ASSERT(__alignof__(celix_properties_iterator_t) == __alignof__(hash_map_iterator_t));
```

I'm OK with the current explicit way, but strongly recommend adding build-time assertions in case of any future inconsistency.